### PR TITLE
Add diagnostic that checks if server directories that should be private are accessible

### PR DESCRIPTION
**Pull request by diosmosis**
_###_ Description:

Adds a new diagnostic that checks that the tmp/ directory is not accessible, and checks that, if present, the .git/ directory is not accessible.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        $label = $this->translator->translate('Diagnostics_RequiredPrivateDirectories');
+
+        $privatePaths = [
+            ['tmp/', 'tmp/empty'], // created by this diagnostic
+            ['.git/', '.git/config'],
+        ];
+
+        // create test file to check if tmp/empty exists
+        Filesystem::mkdir(PIWIK_INCLUDE_PATH . '/tmp');
+        file_put_contents(PIWIK_INCLUDE_PATH . '/tmp/', 'test');
```
Comment (sgiehl) [inline plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php]: should this maybe be

```suggestion
        file_put_contents(PIWIK_INCLUDE_PATH . '/tmp/empty', 'test');
```

File: plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    continue;
+                }
+
+                $testUrl = $baseUrl . $path;
+
+                if (!$isInternetEnabled) {
+                    $unknown = $this->translator->translate('Diagnostics_PrivateDirectoryInternetDisabled', $testUrl) . ' ' . $manualCheck;
+                    $results[] = DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_WARNING, $unknown);
+                    continue;
+                }
```
Comment (sgiehl) [inline plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php]: Does it make sense to show all checks separately in this case? Maybe we could only show one warning and list all directories/files in it instead.

**Review by sgiehl — CHANGES_REQUESTED** [review]

**Comment by Findus23** [comment]
For reference: I also created a similar check in my plugin that also checks for `lang/en.json` (less severe, but indicates that the htaccess doesn't work), `cache/tracker/matomocache_general.php` and `config/config.ini.php` (also showing a warning if it only outputs `;`)

https://github.com/Findus23/matomo-DiagnosticsExtended/blob/main/Diagnostic/URLCheck.php

File: plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php (hunk 2 above)
Comment (diosmosis) [reply plugins/Diagnostics/Diagnostic/RequiredPrivateDirectories.php]: This makes sense, since the setting will always be disabled in the loop

**Comment by diosmosis** [comment]
@sgiehl updated the PR.

Also added the directories @Findus23 mentioned :+1: .

**Review by sgiehl — APPROVED** [review]
Seems to work as expected now, even though the check would be more accurate if the requests would be performed in the browser and not server side.

**Comment by diosmosis** [comment]
Unfortunately I don't think there's an easy way to implement it so an actual browser separate from the server makes those requests. At least none that I can think of.

**Pull request merged**